src/cpp/dynamic-types/DynamicData.cpp

```diff
@@ -256,6 +256,32 @@ void DynamicData::create_members(
     }
 }
 
+size_t DynamicData::get_size() const
+{
+    switch (type_->kind_)
+    {
+        default:
+            return 1;
```

Comment:
If the returned size is the number of elements, then perhaps the method should have a different name ?